pkg/deployment/agent.go

```diff
@@ -120,6 +120,7 @@ func (a *Agent) Get() *appsv1.DaemonSet {
 					Annotations: commonSpec.Annotations,
 				},
 				Spec: corev1.PodSpec{
+					HostNetwork: a.jaeger.Spec.Agent.HostNetWork,
```

Comment:
Once the value is changed to `*bool`, you'll have to nil-check the value and use a default (false).